src/probe/probe_iterator.jl

```diff
@@ -57,19 +67,73 @@ function probe(examples::Vector{<:IOExample}, iterator::ProgramIterator, select:
 
     return nothing
 end
+"""
+    selectpsol_largest_subset(partial_sols::Vector{ProgramCache}) 
 
-@programiterator GuidedSearchIterator(
-    spec::Vector{<:IOExample},
-    symboltable::SymbolTable
-)
+This scheme selects a single cheapest program (first enumerated) that 
+satisfies the largest subset of examples encountered so far across all partial_sols.
+"""
+function selectpsol_largest_subset(partial_sols::Vector{ProgramCache})  
+    if isempty(partial_sols)
+        return []
+    end
+    largest_subset_length = maximum(cache -> length(cache.correct_examples), partial_sols)
+    programs_with_largest_length = filter(cache -> length(cache.correct_examples) == largest_subset_length, partial_sols)
+    # find the program with the smallest cost
+    minimum_cost = minimum(cache -> cache.cost, programs_with_largest_length)
+    best_sol = first(filter(cache -> cache.cost == minimum_cost, programs_with_largest_length))
```

Comment:
You would only need to go through the partial solutions once:
```suggestion
largest_subset_length = 0
cost = typemax(Int)
best_sol = nothing
for psol in partial_sols
    len = length(psol.correct_examples)
    if len > largest_subset_length || len == largest_subset_length && psol.cost < cost
        largest_subset_length = len
        best_sol = psol
        cost = psol.cost
    end
end
```

Reply:
Yes I agree. I will use the for loop approach.

Reply:
I will set `best_sol` to be the first program on the partial_sols instead of nothing to improve avoid an implicit `Union{Nothing,ProgramCache}` that julia will create as the type.

You can inspect this by running `@code_warntype HerbSearch.selectpsol_largest_subset(Vector{ProgramCache}())` in the REPL

![image](https://github.com/Herb-AI/HerbSearch.jl/assets/35890341/de7f30ea-14f8-42af-b771-52f99eb3c0f0)